lib/models/addon.js

```diff
@@ -1445,4 +1442,112 @@ Addon.lookup = function(addon) {
  - [ember-cli-google-analytics](https://github.com/pgrippi/ember-cli-google-analytics/blob/v1.5.0/index.js#L79)
 */
 
+function methodsForTreeType(treeType) {
+  var treeMethods = DEFAULT_TREE_FOR_METHOD_METHODS[treeType];
+
+  return GLOBAL_TREE_FOR_METHOD_METHODS.concat(treeMethods);
+}
+
+if (experiments.ADDON_TREE_CACHING) {
+  addonProto[experiments.ADDON_TREE_CACHING] = function cacheKeyForTree(treeType) {
```

Comment:
This is **absolutely** intended to be used as a canary feature. We discussed this conceptually yesterday  (in person), and the idea is to make `lib/experiments/index.js` simply `module.exports = {};` when we branch to `beta`.  This allows users that want to experiment with the feature to do so on master, but makes it very clear that it is experimental (until https://github.com/ember-cli/rfcs/pull/90 lands).

See a tad more details in [ember-cli/rfcs#91](https://github.com/ember-cli/rfcs/blob/addon-instrumentation-experimental-hooks/active/0000-addon-instrumentation-experimental-hooks.md#experiments)...